src/IntroJs.ts

```diff
@@ -0,0 +1,375 @@
+import mergeOptions from "./util/mergeOptions";
+import exitIntro from "./core/exitIntro";
+import refresh from "./core/refresh";
+import introForElement from "./core/introForElement";
+
+import {
+  populateHints,
+  hideHint,
+  hideHints,
+  showHint,
+  showHints,
+  removeHint,
+  removeHints,
+  showHintDialog,
+} from "./core/hint";
+import {
+  currentStep,
+  goToStep,
+  goToStepNumber,
+  nextStep,
+  previousStep,
+} from "./core/steps";
+
+
+export interface IntroItem {
```

Comment:
I personally love this change to typescript. However I would advice to put each interface in it's own file and let the name end with Interface or start with an extra I, so people know it's not a class but an interface, and they can more easily re-use it outside.

Also checkout this blog article, this way you can compile it as both an ejs and an mjs module
https://www.sensedeep.com/blog/posts/2021/how-to-create-single-source-npm-module.html

Reply:
This does not make any sense, the typescript create interfaces in the dist folder